packages/data-connect/src/api/DataConnect.ts

```diff
@@ -191,6 +194,16 @@ export class DataConnect {
   }
 }
 
+// @internal
+export function areTransportOptionsEqual(
+  transportOptions1: TransportOptions,
+  transportOptions2: TransportOptions
+) {
+  return (
+    JSON.stringify(transportOptions1) === JSON.stringify(transportOptions2)
```

Comment:
This is not the best test for deep equality -- it returns false if the key orders are different, for example. Up to you whether to fix it or just add a TODO though.

Reply:
Ah, thanks for catching this. I might just do a prop equality then. I was iffy on that since we have to update the function if we add more fields to `TransportOptions`, but that seems like it's not a block-worthy tradeoff. Making the change.